lib/ruby_indexer/lib/ruby_indexer/visitor.rb

```diff
@@ -25,6 +26,13 @@ def visit_class_node(node)
       add_class_entry(node)
     end
 
+    sig { override.params(node: Prism::SingletonClassNode).void }
+    def visit_singleton_class_node(node)
+      @singleton_class = true
```

Comment:
What about usages of:
```ruby
class << Foo
  def bar
  end
end
```
syntax? That opens up the singleton class of `Foo` and the scope inside defines stuff on there.

So, I think this flag shouldn't a boolean, but store the name of the singleton class that's opened (which should be accessible via `node.expression`) or `nil`.

Reply:
👍 Updated, although for now I'm storing the node rather than the name of the class, in case we need anything from that. Once we get further along in https://github.com/Shopify/ruby-lsp/pull/1121 I'll have better understanding of if that's necessary.